proposals/stf-general-investment.md

```diff
@@ -0,0 +1,263 @@
+  - This limits our ability to do long-term planning, and many things can't be done without breaking changes
+- Overall, we have very little visibility into what things cost in terms of volunteer time and effort
+- We currently don’t know how to self-host large repositories such as Nixpkgs, and we’re stuck to GitHub
+
+## What are possible alternatives to your project and how does your project compare to them? (optional) 300 words
+
+- Guix
+
+  Guix is an early fork of Nix using Guile as a configuration and implementation language, and is the only project comparable to the Nix ecosystem.
+  Guix provides its own software distribution and operating system configuration framework, with free software exclusively.
+  The community is much smaller, the package set is rather focused on scientific applications.
+  Guix pioneered full-source bootstrapping, which is now ported to Nixpkgs.
+
+- Arch Linux
+
+  As a software distribution, it has many similarities to Nixpkgs and NixOS in terms of size, structure, and procedures.
+  It does not feature some of the key properties of the Nix ecosystem, such as declarative, composable configurations through uniform functional interfaces.
+  The Arch community has advanced the state of the art in [reproducible builds](https://reproducible-builds.org) in the past, and inspired similar work in Nixpkgs.
+
+## What do you plan to implement with the support from STF? 900 words
+
+*Please describe your objectives and the corresponding activities.
+State clearly how they contribute to the improvement or maintenance of the technology, especially with regard to security and resilience.*
+
+We understand resilience as the ability to recover from disruptions and to adapt to a changing environment.
+Apart from implementing technical measures to that end, merely having qualified engineers available for ongoing maintenance and addressing small-scale, long-term, or cross-cutting issues as part of their engagement will improve the overall health of the ecosystem by unblocking volunteer efforts more quickly than is currently possible.
+We expect this to have a compounding effect on overall productivity, as we estimate that more than 95% of the work is currently done by volunteers, many of whom are slowed down by lack of support and swift decision making.
+
+- Improve the contributor experience (300k)
+
+  Streamlining the contribution process should make it more attractive for newcomers to participate in development, and keep existing people around and motivated.
+  Growing the contributor community will make the ecosystem more robust against fluctuation.
+  We want to achieve this by making more systemic knowledge explicit and easier to discover, access, and manipulate, and thus ease onboarding of new contributors, and reduce contribution turnaround time.
+
+  - Tools for review automation (50k)
+  - Modularise the Nix configuration system (30k)
```

Comment:
What part should be modularised here? Because NixOS is about as modular as it gets.

Reply:
This is a compound item that comprises all of the code reorganisation tasks from [this discussion with Nix maintainers](https://discourse.nixos.org/t/2024-06-05-nix-team-meeting-minutes-150/46583), mostly revolving around https://github.com/NixOS/nix/issues/5638.

The code that eventually parses `nix.conf` and command line arguments is a big swamp of global variables and static initialisation. A bunch of people, including myself, have collectively spent dozens of hours on [fixing one single bug in there](https://github.com/NixOS/nix/pull/11079). It's a huge time sink for maintainers and contributors.